girdermedviewer/app/girder/utils.py

```diff
@@ -42,22 +43,37 @@ def __del__(self):
     def get_item_files(self, item):
         return self.girder_client.listFile(item["_id"])
 
-    @contextmanager
     def download_file(self, file):
-        """
-        :param forced_cache can be used to override the cache mode for a specific file
-        """
-        file_path = os.path.join(self.temp_dir_path, file["_id"], file["name"])
+        file_path = os.path.join(self.temp_dir_path, file['_id'], file["name"])
         if not os.path.exists(file_path):
             logger.info(f"Download {file['name']} to {file_path}")
             self.girder_client.downloadFile(
                 file["_id"],
                 file_path
             )
+        return file_path
+
+    @contextmanager
+    def fetch_file(self, file):
+        fetched_from_assetstore = False
+        clear_cache_required = False
+        if self.assetstore_dir_path:
+            if "path" not in file:
+                logger.info("The file path in assetstore is unknown. Make sure to use the girdermedviewer-plugin")

```

Comment:
Should probably raise an exception instead. Because next line will raise an exception `file['path']`.